packages/x-data-grid/src/components/cell/GridCell.tsx

```diff
@@ -66,12 +66,19 @@ export type GridCellProps = {
   gridHasFiller: boolean;
   onClick?: React.MouseEventHandler<HTMLDivElement>;
   onDoubleClick?: React.MouseEventHandler<HTMLDivElement>;
+  onMouseEnter?: React.MouseEventHandler<HTMLDivElement>;
   onMouseDown?: React.MouseEventHandler<HTMLDivElement>;
+  onMouseLeave?: React.MouseEventHandler<HTMLDivElement>;
   onMouseUp?: React.MouseEventHandler<HTMLDivElement>;
+  onMouseOver?: React.MouseEventHandler<HTMLDivElement>;
+  onKeyUp?: React.KeyboardEventHandler<HTMLDivElement>;
   onKeyDown?: React.KeyboardEventHandler<HTMLDivElement>;
   onDragEnter?: React.DragEventHandler<HTMLDivElement>;
   onDragOver?: React.DragEventHandler<HTMLDivElement>;
-  [x: string]: any; // TODO v7: remove this - it breaks type safety
```

Comment:
This looks like a slight BC (however, technically I'd treat it as a (loophole)bug-fix). 🙈 
Similar case like the toolbar .

Reply:
Well, it's still possible to pass all the HTML attributes that `div` can accept + `data-` attributes.
It'll create type errors for custom props though, in which case type augmentation should be used: https://mui.com/x/react-data-grid/components/#custom-slot-props-with-typescript

Reply:
> It'll create type errors for custom props though, in which case type augmentation should be used: https://mui.com/x/react-data-grid/components/#custom-slot-props-with-typescript

Yes, these changes basically uncovered the loophole that was present. 🙈 
Have you tried using module augmentation in those demos to better showcase the usage of custom slots without the need for `as` casting?

Reply:
Which demos?